lib/views/pages/home/posts/home_feed.dart

```diff
@@ -8,48 +10,94 @@ import "package:proxima/views/sort_option_widgets/feed_sort_option/feed_sort_opt
 /// It contains the posts
 class HomeFeed extends HookConsumerWidget {
   static const feedSortOptionKey = Key("feedSortOption");
-  static const homeFeedKey = Key("homeFeed");
   static const emptyHomeFeedKey = Key("emptyHomeFeed");
+  static const refreshButtonKey = Key("refreshButton");
   const HomeFeed({super.key});
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final posts = ref.watch(postList);
+    final asyncPosts = ref.watch(postOverviewProvider);
 
     final emptyHelper = Center(
       key: emptyHomeFeedKey,
-      child: Row(
+      child: Column(
         mainAxisAlignment: MainAxisAlignment.center,
         children: [
-          const Text("No post to display, "),
-          InkWell(
-            onTap: () => {
-              //TODO: Add navigation to create post page
-            },
-            child: const Text(
-              "create one",
-              style: TextStyle(
-                color: Colors.blue,
+          Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              const Text("No post to display, "),
+              InkWell(
+                onTap: () => {
+                  //TODO: Add navigation to create post page
+                },
+                child: const Text(
+                  "create one",
+                  style: TextStyle(
+                    color: Colors.blue,
+                  ),
+                ),
               ),
-            ),
+            ],
+          ),
+          const SizedBox(height: 10),
+          ElevatedButton(
+            key: refreshButtonKey,
+            onPressed: () async {
+              ref.read(postOverviewProvider.notifier).refresh();
+            },
+            child: const Text("Refresh"),
           ),
         ],
       ),
     );
 
-    final postsCards = ListView(
-      key: homeFeedKey,
-      children: posts.map((post) => PostCard(post: post)).toList(),
-    );
-
     return Column(
       children: [
         const FeedSortOptionChips(
           key: feedSortOptionKey,
         ),
         const Divider(),
-        Expanded(child: posts.isEmpty ? emptyHelper : postsCards),
+        CircularValue(
+          value: asyncPosts,
+          builder: (context, posts) {
+            return PostList(
+              emptyHelper: emptyHelper,
+              posts: posts,
+              onRefresh: () async {
+                return ref.read(postOverviewProvider.notifier).refresh();
+              },
+            );
+          },
+        ),
       ],
     );
   }
 }
+
+class PostList extends StatelessWidget {
+  const PostList({
+    super.key,
+    required this.emptyHelper,
+    required this.posts,
+    required this.onRefresh,
+  });
+  static const homeFeedKey = Key("homeFeed");
+
+  final Center emptyHelper;
```

Comment:
The code should never assume the type of Widget one passes as argument, as it might as well change in the future. This should be of type `Widget`.

By the way, it is not very good practice to pass helper widgets around like that and encapsulate logic that is not managed by the current Widget agent.
This code should leave the empty out of the `PostList` and handle the empty case in the parent (`HomeFeed`) widget.

Reply:
Fixed by 729756e